app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -289,7 +289,7 @@ public boolean addOldMessage(Message message, int messageAndHeadersCount, String
      */
     public void addNewMessage(Message message) {
         MessageHeaderParent item = null;
-        for (int i = getItemCount(false) - 1; i > 1; i--) {
+        for (int i = getItemCount(false) - 1; i >= 1; i--) {
```

Comment:
Why not?

Firstly, I tried the code and it works fine for me. The messages are added to the bottom and I couldn't find any bug during my testing.

And even in principle, I don't see why messages would not be added at the bottom of the list. The messages are added using `items.add(getItemCount(true) - 1, message)`, which does not use the value of i anywhere. I don't know why one more iteration of the for loop would mess with the message adding part. Earlier, the MessageHeaderParent at the top wasn't accounted for. With i going equal to 1, that is fixed.

I don't know why you think this is a problem. Did you try running the code? If you are seeing something wrong when testing, please let me know, and I'll try on other devices. At least on my device, the code works fine.